### PR TITLE
feat: add dbt models for ADP Payroll [DE-412]

### PR DESCRIPTION
**Pull request by emily-flambe**
cc @royconst, all we need to do now is add some dbt operators to the Payroll DAG and these models will do the following:

- starting with `_raw_payroll`, generate an ephemeral model casting all the fields to appropriate datatypes. Note that for the fields that reflect dollar values, I took the liberty of chopping off the leading `$ ` and converting them to floats, and appending `_USD` to the field names. This is how _serious people_ store money-data in a database (so that they can be treated as numbers you know?)
- update incremental model `payroll_base` using the date-partition-replacement strat we use for tables that run a full refresh each day (note that this means this dbt should run every day regardless of whether new data was ingested in the sync).
- materialize `payroll_base` into a matview in the partner project called `payroll` that runs `SELECT * from payroll_base`.

This PR also includes a schema yaml to run a single dbt test that checks for not-null and unique on the _hashid field in `payroll_base` that was calculated by hashing the combo of all of the non-dollar-amount fields present in `_raw_payroll`.

This dbt runs locally and you can see the _base table and the partner matview in BigQuery! I will issue a separate PR that will add the dbt tasks to the ADP Payroll DAG.

### CODE REVIEW

**Comment by emily-flambe** [comment]
(converted this to draft because I remembered this table updates incrementally, not full refresh! oops)

**Comment by emily-flambe** [comment]
okay, NOW this is ready for review - when there's a good time that is (so maybe next week lol) @royconst @jitoquinto @kanelouise

**Comment by emily-flambe** [comment]
@jitoquinto @kanelouise @royconst - per my recent post in slack, this dbt now prepends `_` to three field names that start with a number in the ADP Payroll source data. RIP my afternoon lmao.

File: dbt-cta/adp_payroll/models/1_cta_incremental/payroll_base.sql
[the remark this answers is not in the document]
Comment (emily-flambe) [reply dbt-cta/adp_payroll/models/1_cta_incremental/payroll_base.sql:1]: ope so these models should be run incrementally, not with date partition replacement - so if I have any references to that remaining then I shall remove them! (My other recent dbt PR for Openfield _was_ full refresh overwrite.)

My understanding is that running this in `backfill` mode should be something do either zero times or one time, hopefully - the goal is to set this and forget it. If we do backfill, then I think we'd probably want to just truncate the payroll table entirely before running the DAG in backfill. If we find ourselves doing that more than once then it means something is wrong with the world lol

File: dbt-cta/adp_payroll/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# ADP Payroll
+
+These dbt models do the following for ADP Payroll data (synced via Airflow from SFTP into BigQuery):
+
+1. starting with _raw_payroll, generate an ephemeral model casting all the fields to appropriate datatypes. Note that for the fields that reflect dollar values, I took the liberty of chopping off the leading $  and converting them to floats, and appending _USD to the field names. This is how serious people store money-data in a database (so that they can be treated as numbers you know?)
+2. update incremental model payroll_base using the date-partition-replacement strat we use for tables that run a full refresh each day (note that this means this dbt should run every day regardless of whether new data was ingested in the sync).
```
Comment (emily-flambe) [inline dbt-cta/adp_payroll/README.md:6]:
```suggestion
2. update incremental model payroll_base (note that this means this dbt should run every day regardless of whether new data was ingested in the sync).
```

File: dbt-cta/adp_payroll/models/1_cta_incremental/payroll_base.sql
[the remark this answers is not in the document]
Comment (jitoquinto) [reply dbt-cta/adp_payroll/models/1_cta_incremental/payroll_base.sql:1]: Ah okay thanks! And that makes sense to me. I think it would be helpful to write that bit down of, if were doing backfill for this Dag you should truncate the table so we dont forget if we do need to do that!

**Review by jitoquinto — APPROVED** [review]
My one concern has been answered and the rest lgtm! Glad people will be able to do all the money maths they want now 💸

Left a comment to your response just asking if we can document somewhere (probably on the dag and not the dbt) that we should be truncating in the event we have to run a backfill

**Comment by emily-flambe** [comment]
> My one concern has been answered and the rest lgtm! Glad people will be able to do all the money maths they want now 💸
> 
> Left a comment to your response just asking if we can document somewhere (probably on the dag and not the dbt) that we should be truncating in the event we have to run a backfill

I added notes about Backfill mode (including the advisory to truncate the _base table) in the airflow-dags PR: https://github.com/community-tech-alliance/airflow-dags/pull/496/commits/608cd0754b251f1dcaee6c113899478192021b33

Gonna merge this one 🎉
